server/market/orderrouter.go

```diff
@@ -452,6 +475,50 @@ func (r *OrderRouter) extractMarket(prefix *msgjson.Prefix) (MarketTunnel, *msgj
 	return tunnel, nil
 }
 
+// SuspendEpoch holds the index and end time of final epoch marking the
+// suspension of a market.
+type SuspendEpoch struct {
+	Idx int64
+	End time.Time
+}
+
+// SuspendMarket schedules a suspension of a given market, with the option to
+// persist the orders on the book (or purge the book automatically on market
+// shutdown). The scheduled final epoch and suspend time are returned. Note that
+// OrderRouter is a proxy for this request to the ultimate Market. This is done
+// because OrderRouter is the entry point for new orders into the market. TODO:
+// track running, suspended, and scheduled-suspended markets, appropriately
+// blocking order submission according to the schedule rather than just checking
+// Market.Running prior to submitting incoming orders to the Market.
+func (r *OrderRouter) SuspendMarket(mktName string, asSoonAs time.Time, persistBooks bool) *SuspendEpoch {
+	mkt, found := r.tunnels[mktName]
+	if !found {
+		return nil
+	}
+
+	idx, t := mkt.Suspend(asSoonAs, persistBooks)
+	return &SuspendEpoch{
+		Idx: idx,
+		End: t,
+	}
+}
+
+func (r *OrderRouter) Suspend(asSoonAs time.Time, persistBooks bool) map[string]*SuspendEpoch {
```

Comment:
Not used presently, but there is likely a need for a "suspend all" with a time rather than immediate sigint shutdown.